### PR TITLE
Separate PyNEST API modules into separate pages in documentation

### PR DESCRIPTION
**Pull request by jessica-mitchell**
This PR separates the modules for the API into separate pages, so there is no longer a single long list of all functions.
Tried to order modules so more  useful ones near top.  ll_api is not included. 

See example output here: https://nest-simulator--2588.org.readthedocs.build/en/2588/ref_material/pynest_api/index.html

### CODE REVIEW

**Review by clinssen — CHANGES_REQUESTED** [review]
Hi, sorry for the latency! The version rendered on nest-test.readthedocs.org seems to be outdated, could you re-render? Here are some initial thoughts:

- "nest" in "nest module" is sometimes typeset in italics and sometimes in a fixed width font. Can we make it a clickable link to the API documentation for the nest module?
- Parameter types are not typeset propertly in autodoc ("Type:" does not appear at the beginning of the line; there seem to be line breaks missing)
- Similar misalignment between "Parameters" and "Notes" headings on Run(t)(https://nest-test.readthedocs.io/en/separate-api/ref_material/pynest_api/nest.lib.hl_api_simulation.html#nest.lib.hl_api_simulation.Run)
- There is a header "ll_api = <module 'nest.ll_api' from '/home/docs/checkouts/readthedocs.org/user_builds/nest-test/checkouts/separate-api/pynest/nest/ll_api.py'>[¶](https://nest-test.readthedocs.io/en/separate-api/ref_material/pynest_api/kernel_attributes.html#nest.NestModule.ll_api)" on the page ``kernel_attributes.html``. This seems incorrect; can we eliminate the path here?
- Deprecation warnings are not rendered (@deprecated decorator)
- Double quotes delineating strings are rendered as single quotes

File: doc/htmldoc/templates/index.html
Comment (jessica-mitchell) [inline doc/htmldoc/templates/index.html:104]:
```suggestion
            <li><a href="ref_materia/lpynest_api/nest.lib.hl_api_nodes.html">Get API docs for Create</a></li>
```

**Comment by med-ayssar** [comment]
The [Kernel attributes ](https://nest-test.readthedocs.io/en/separate-api/ref_material/pynest_api/kernel_attributes.html) is a bit broken:
The types are **floating** in the middle of the block, and the `raster_plot` module is shown as the absolute path (see photo):

Both issues are depicted in the photo:

Screenshot from 2023-03-21 13-28-09

**Comment by clinssen** [comment]
Looks very good! Two questions:

Why does the text "class NestModule(name)" show up in one spot?

As the categories are now manually curated (if I understand properly), this would imply that when the API changes, this page needs to be changed manually again. I just want to double-check that there is no way to generate this page fully (or almost fully) automatically.

**Review by clinssen — APPROVED** [review]

**Comment by jougs** [comment]
[I'm taking over @med-ayssar's request for review, so sorry for being late to the party]

I am generally not sure if this split is a good idea for multiple reasons:

1. The [new index](https://nest-test.readthedocs.io/en/separate-api/ref_material/pynest_api/index.html) mainly consists of (IMO way too large) headlines that basically just repeat what the (often single) bullet points below them already say. Because of this redundancy, the information content of the page is really low and none of the new sections provides any guidance for when to click which of the links.
2. Imagine a situation where the user coarsely knows that there is a function called `somethingStatus` for setting neuron parameters and they want to find the getter counterpart for that. On the old page they could just hit `Ctrl+F` and type `status` and get enlightened in no time. How would this work in the new structure? 
3. This new index somehow suggests that *Node module*, *Models module*, and all these other things are stand-alone entities or sub-modules of PyNEST that you need to import manually to get their functionality, while in reality they are merely reflecting the internal structure of the code that does a) not necessarily map to the user's intended workflow in a meaningful way and might b) be already a bad idea on the code level
4. The page mixes things on different semantic levels without explaining this
   - *Kernel Attributes* are fundamental settings for the simulation kernel
   - The *bla Module*s provide the functions and classes of the API
   - The *Functions related to visual elements* are high-level helpers meant to spare the user from re-implementing boring plotting tasks in their simulations. They are not really part of the API  
5. The rendering of the kernel attributes is still broken as [pointed out](https://github.com/nest/nest-simulator/pull/2588#issuecomment-1477755066) above. Also the *NestModule*, *ll_api = <module 'nest.ll_api'* and similar things  contained on that page are implementation details the user should not care about.
6. The way the different sections [come into existence](https://github.com/nest/nest-simulator/pull/2588/files#diff-df88e38762870284a955889613ba33ce349683e64e824f5a3e18187ab6ec454bR12-R28) looks crazy repetitive and not very maintainable to me.

I think we need to discuss this in person to arrive at a solution that actually improves the accessibility of the API reference.

**Comment by jessica-mitchell** [comment]
@jougs sorry I think there is an old link there - and what you looked at is not the most recent changes
See here:https://nest-test.readthedocs.io/en/split-api-docs/ref_material/pynest_api/index.html

**Comment by jougs** [comment]
Oh! In this case forget everything I said before. The new version of the page looks indeed wonderful!

# :heart_eyes_cat: 

I will try to do a more detailed review very soon. Stay tuned!

File: doc/htmldoc/ref_material/pynest_api/index.rst
Comment (jougs) [inline doc/htmldoc/ref_material/pynest_api/index.rst:309]: Ideally, I'd like to see the attributes on the same level as the functions in the other sections, without the `class NestModule(name)`  header. Obviously ignore this request for change if implementing it would mean that all of them would have to be listed individually or something like that.

Screenshot from 2023-08-15 11-00-41

**Review by jougs — COMMENTED** [review]
Very nice indeed. I only have some (minor) comments on code duplication and design and layout.

File: doc/requirements.txt
Comment (jessica-mitchell) [reply doc/requirements.txt:10]: this is so you can access the API docs for nest.server.hl_api_server

Comment (jougs) [reply doc/requirements.txt:10]: :+1:

File: doc/htmldoc/ref_material/pynest_api/index.rst
Comment (jessica-mitchell) [reply doc/htmldoc/ref_material/pynest_api/index.rst]: helper modules are gone

File: doc/htmldoc/ref_material/pynest_api/index.rst
Comment (jessica-mitchell) [reply doc/htmldoc/ref_material/pynest_api/index.rst]: fixed

File: doc/htmldoc/static/css/custom.css
Comment (jessica-mitchell) [reply doc/htmldoc/static/css/custom.css:77]: fixed

File: doc/htmldoc/ref_material/pynest_api/index.rst
Comment (jessica-mitchell) [reply doc/htmldoc/ref_material/pynest_api/index.rst]: NestModule line removed, indentation removed

**Review by jougs — APPROVED** [review]
Many many thanks for your endurance on this one and the great work on massively improving things. I totally approve and merge without second review.

**Pull request merged**
